### PR TITLE
Fix some unnecessary use of boxed types and incorrect format strings spotted by lgtm.

### PR DESCRIPTION
**Pull request by chrisgavin**
This pull request fixes some small problems identified by lgtm. You can find [more details here](https://lgtm.com/projects/g/druid-io/druid/alerts/).

The first commit changes some boxed types which can never be null to unboxed types.

The second commit fixes some format strings where the number of arguments doesn't match the number of placeholders in the format string.

### CODE REVIEW

**Comment by jihoonson** [comment]
@chrisgavin thanks for your patch. Please check the compilation error.

**Comment by chrisgavin** [comment]
Thanks. I have fixed it. :)

**Comment by leventov** [comment]
@chrisgavin please in the future, don't amend commits in your PR, always add new commits on top, including merge commits.

**Comment by chrisgavin** [comment]
@leventov Sorry about that. I could revert to the previous HEAD of this PR and then make the compilation fix as a new commit if that would be helpful.

**Comment by leventov** [comment]
@chrisgavin not needed now, it's for the future.

Also could you please add inspections for this kind of errors?

1) Add to Druid.xml:
```
    <inspection_tool class="MalformedFormatString" enabled="true" level="ERROR" enabled_by_default="true">
      <option name="additionalClasses" value="io.druid.java.util.common.StringUtils,io.druid.java.util.common.logger.Logger" />
      <option name="additionalMethods" value="trace,debug,info,warn,error,wtf,format,nonStrictFormat" />
    </inspection_tool>
```

2) Add to checkstyle.xml:
```
    <module name="Regexp">
      <property name="format" value="com\.metamx\.common\.logger"/>
      <property name="illegalPattern" value="true"/>
      <property name="message" value="Use io.druid.java.util.common.logger.Logger instead"/>
    </module>
```

3) I cannot find an inspection for unecessary use for boxed types (there is "unnecessary boxing" but it doesn't spot the places that you have found), but maybe there is some.

**Comment by leventov** [comment]
Or more generally, you may prohibit any non-druid loggers by regex like
```
(?<!io.druid.java.util.common.logger)\.Log(ger)?[^A-Za-z0-9]
```

But a few exceptions will be needed, that you may guard with `//CHECKSTYLE.OFF: Regexp` and `//CHECKSTYLE.ON: Regexp` comments.

**Comment by leventov** [comment]
@chrisgavin it wasn't supposed to be clean, please fix inspection and checkstyle failures now.

File: extensions-core/kafka-indexing-service/src/test/java/io/druid/indexing/kafka/KafkaIndexTaskTest.java [hunk 1]
```diff
@@ -37,7 +37,9 @@
 import com.google.common.util.concurrent.ListenableFuture;
 import com.google.common.util.concurrent.ListeningExecutorService;
 import com.google.common.util.concurrent.MoreExecutors;
+//CHECKSTYLE.OFF: Regexp
```
Comment (leventov) [inline extensions-core/kafka-indexing-service/src/test/java/io/druid/indexing/kafka/KafkaIndexTaskTest.java]: Why needs to use old logger?

File: indexing-service/src/test/java/io/druid/indexing/common/task/RealtimeIndexTaskTest.java [hunk 2]
```diff
@@ -31,7 +31,9 @@
 import com.google.common.util.concurrent.ListenableFuture;
 import com.google.common.util.concurrent.ListeningExecutorService;
 import com.google.common.util.concurrent.MoreExecutors;
+//CHECKSTYLE.OFF: Regexp
```
Comment (leventov) [inline indexing-service/src/test/java/io/druid/indexing/common/task/RealtimeIndexTaskTest.java]: Why needs to use old logger?

File: java-util/src/test/java/io/druid/java/util/common/logger/LoggerTest.java [hunk 3]
```diff
@@ -31,6 +31,7 @@ public void testLogWithCrazyMessages()
     log.warn(message);
   }
 
+  /** @noinspection MalformedFormatString */
```
Comment (leventov) [inline java-util/src/test/java/io/druid/java/util/common/logger/LoggerTest.java]: Use SuppressWarnings

File: server/src/test/java/io/druid/segment/realtime/appenderator/AppenderatorTester.java [hunk 4]
```diff
@@ -21,7 +21,9 @@
 
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.collect.ImmutableMap;
+//CHECKSTYLE.OFF: Regexp
```
Comment (leventov) [inline server/src/test/java/io/druid/segment/realtime/appenderator/AppenderatorTester.java]: Why needs to use old logger?

File: server/src/test/java/io/druid/server/coordinator/rules/LoadRuleTest.java [hunk 5]
```diff
@@ -28,7 +28,9 @@
 import com.google.common.collect.Sets;
 import com.google.common.util.concurrent.ListeningExecutorService;
 import com.google.common.util.concurrent.MoreExecutors;
+//CHECKSTYLE.OFF: Regexp
```
Comment (leventov) [inline server/src/test/java/io/druid/server/coordinator/rules/LoadRuleTest.java]: Why needs to use old logger?

**Comment by chrisgavin** [comment]
@leventov The places that still use the old logger do so because the logger is passed into a `LoggingEmitter` constructor, which only accepts a metamx logger.

**Comment by leventov** [comment]
@chrisgavin 16 more malformed string format violations reported: https://teamcity.jetbrains.com/viewLog.html?buildId=1108816&tab=Inspection&buildTypeId=OpenSourceProjects_Druid_InspectionsPullRequests

File: processing/src/main/java/io/druid/query/QueryRunnerHelper.java [hunk 6]
```diff
@@ -64,7 +64,7 @@
               @Override
               public Result<T> apply(Cursor input)
               {
-                log.debug("Running over cursor[%s]", adapter.getInterval(), input.getTime());
+                log.debug("Running over cursor[%s]", input.getTime());
```
Comment (leventov) [inline processing/src/main/java/io/druid/query/QueryRunnerHelper.java]: Strange logging "cursor[input.getTime()]". Also maybe just remove this debugging line

File: processing/src/main/java/io/druid/query/groupby/epinephelinae/AbstractBufferGrouper.java [hunk 7]
```diff
@@ -195,7 +195,7 @@ public void close()
         aggregator.close();
       }
       catch (Exception e) {
-        log.warn(e, "Could not close aggregator, skipping.", aggregator);
+        log.warn(e, "Could not close aggregator, skipping.");
```
Comment (leventov) [inline processing/src/main/java/io/druid/query/groupby/epinephelinae/AbstractBufferGrouper.java]: Why not print the aggregator?

File: processing/src/main/java/io/druid/query/topn/TopNQueryEngine.java [hunk 8]
```diff
@@ -94,7 +94,7 @@ public TopNQueryEngine(NonBlockingPool<ByteBuffer> bufferPool)
               @Override
               public Result<TopNResultValue> apply(Cursor input)
               {
-                log.debug("Running over cursor[%s]", adapter.getInterval(), input.getTime());
+                log.debug("Running over cursor[%s]", input.getTime());
```
Comment (leventov) [inline processing/src/main/java/io/druid/query/topn/TopNQueryEngine.java]: Same as above

**Comment by leventov** [comment]
@chrisgavin also please fill the CLA http://druid.io/community/cla.html

**Pull request merged**